src/kernels/kernelStartupCodeProvider.node.ts

```diff
@@ -2,80 +2,43 @@
 // Licensed under the MIT License.
 
 import { inject, injectable } from 'inversify';
-import { Uri } from 'vscode';
-import { IWorkspaceService } from '../platform/common/application/types';
 import { CodeSnippets } from '../platform/common/constants';
 import { getFilePath } from '../platform/common/platform/fs-paths';
-import { IFileSystem } from '../platform/common/platform/types';
-import { IConfigurationService } from '../platform/common/types';
-import { calculateWorkingDirectory } from '../platform/common/utils.node';
 import { traceInfo } from '../platform/logging';
 import { isLocalHostConnection, isPythonKernelConnection } from './helpers';
-import { expandWorkingDir } from './jupyter/jupyterUtils';
+import { KernelWorkingFolder } from './kernelWorkingFolder.node';
 import { IKernel, isLocalConnection, IStartupCodeProvider, StartupCodePriority } from './types';
 
 @injectable()
 export class KernelStartupCodeProvider implements IStartupCodeProvider {
     public priority = StartupCodePriority.Base;
 
-    constructor(
-        @inject(IConfigurationService) private readonly configService: IConfigurationService,
-        @inject(IFileSystem) private readonly fs: IFileSystem,
-        @inject(IWorkspaceService) private readonly workspaceService: IWorkspaceService
-    ) {}
+    constructor(@inject(KernelWorkingFolder) private readonly kernelWorkingFolder: KernelWorkingFolder) {}
 
     async getCode(kernel: IKernel): Promise<string[]> {
-        // If this is a live kernel, we shouldn't be changing anything by running startup code.
+        // We cannot change paths for remote live kernels.
+        if (kernel.kernelConnectionMetadata.kind === 'connectToLiveRemoteKernel') {
+            return [];
+        }
+        // Only local kernels are supported.
         if (
-            !isPythonKernelConnection(kernel.kernelConnectionMetadata) &&
-            kernel.kernelConnectionMetadata.kind !== 'connectToLiveRemoteKernel'
+            !isLocalConnection(kernel.kernelConnectionMetadata) &&
+            !isLocalHostConnection(kernel.kernelConnectionMetadata)
         ) {
             return [];
         }
-        if (!isLocalConnection(kernel.kernelConnectionMetadata)) {
+        if (!isPythonKernelConnection(kernel.kernelConnectionMetadata)) {
             return [];
         }
 
-        if (
-            isLocalConnection(kernel.kernelConnectionMetadata) ||
-            isLocalHostConnection(kernel.kernelConnectionMetadata)
-        ) {
-            let suggestedDir = await calculateWorkingDirectory(
-                this.configService,
-                this.workspaceService,
-                this.fs,
-                kernel.resourceUri
-            );
-            if (suggestedDir && (await this.fs.exists(suggestedDir))) {
```

Comment:
As you can see we have more complex logic here and its partially split into two files.
With this PR its all in 1 place with tests.